beacon/message.go

```diff
@@ -17,31 +17,35 @@ type ProposalVrfMessage struct {
 type ProposalMessage struct {
 	EpochID      types.EpochID
 	NodeID       types.NodeID
-	VRFSignature []byte
+	VRFSignature []byte `scale:"max=80"`
+}
+
+type Proposal struct {
+	Value []byte `scale:"max=32"` // TODO(mafa): check if max. value is correct
 }
 
 // FirstVotingMessageBody is FirstVotingMessage without a signature.
 type FirstVotingMessageBody struct {
 	EpochID                   types.EpochID
-	ValidProposals            [][]byte
-	PotentiallyValidProposals [][]byte
+	ValidProposals            []Proposal `scale:"max=32"` // TODO(mafa): check if max. value is correct
+	PotentiallyValidProposals []Proposal `scale:"max=32"` // TODO(mafa): check if max. value is correct
 }
 
 // FirstVotingMessage is a message type which is used when sending first voting messages.
 type FirstVotingMessage struct {
 	FirstVotingMessageBody
-	Signature []byte
+	Signature []byte `scale:"max=64"`
 }
 
 // FollowingVotingMessageBody is FollowingVotingMessage without a signature.
 type FollowingVotingMessageBody struct {
 	EpochID        types.EpochID
 	RoundID        types.RoundID
-	VotesBitVector []byte
+	VotesBitVector []byte `scale:"max=32"` // TODO(mafa): check if max. value is correct
```

Comment:
the protocol guarantees that there are very low probability that there will be over 100 proposals.
so a bit vector of 100 bits is 13 bytes. i think 16 byte may suffice.